runtime/src/main/java/org/corfudb/runtime/clients/BaseHandler.java

```diff
@@ -0,0 +1,156 @@
+     *
+     * @param msg The ping request message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return Always True, since the ping message was successful.
+     */
+    @ClientHandler(type = CorfuMsgType.PONG)
+    private static Object handlePong(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return true;
+    }
+
+    /**
+     * Handle an ACK response from the server.
+     *
+     * @param msg The ping request message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return Always True, since the ACK message was successful.
+     */
+    @ClientHandler(type = CorfuMsgType.ACK)
+    private static Object handleAck(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return true;
+    }
+
+    /**
+     * Handle a NACK response from the server.
+     *
+     * @param msg The ping request message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return Always True, since the ACK message was successful.
+     */
+    @ClientHandler(type = CorfuMsgType.NACK)
+    private static Object handleNack(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return false;
+    }
+
+    /**
+     * Handle a WRONG_EPOCH response from the server.
+     *
+     * @param msg The wrong epoch message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return none, throw a wrong epoch exception instead.
+     */
+    @ClientHandler(type = CorfuMsgType.WRONG_EPOCH)
+    private static Object handleWrongEpoch(CorfuPayloadMsg<Long> msg, ChannelHandlerContext ctx,
+                                           IClientRouter r) {
+        throw new WrongEpochException(msg.getPayload());
+    }
+
+    /**
+     * Handle a Version response from the server.
+     *
+     * @param msg The version message
+     * @param ctx The context the message was sent under
+     * @param r   A reference to the router
+     * @return The versioninfo object.
+     */
+    @ClientHandler(type = CorfuMsgType.VERSION_RESPONSE)
+    private static Object handleVersionResponse(JSONPayloadMsg<VersionInfo> msg,
+                                                ChannelHandlerContext ctx, IClientRouter r) {
+        return msg.getPayload();
+    }
+
+    @ClientHandler(type = CorfuMsgType.NOT_READY)
+    private static Object handleNotReady(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        throw new ServerNotReadyException();
+    }
+
+    /** Generic handler for a server exception. */
+    @ClientHandler(type = CorfuMsgType.ERROR_SERVER_EXCEPTION)
+    private static Object handleServerException(CorfuPayloadMsg<ExceptionMsg> msg,
+                                                ChannelHandlerContext ctx, IClientRouter r)
+        throws Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)